torch/_dynamo/eval_frame.py

```diff
@@ -115,14 +115,17 @@ def _initialize(self):
             self.forward = self._call_lazy_check
 
     def __getstate__(self):
-        state = dict(self.__dict__)
-        state.pop("forward", None)
-        state.pop("__call__", None)
-        return state
+        return {"_orig_mod": self._orig_mod}
 
     def __setstate__(self, state):
-        self.__dict__ = state
-        self._initialize()
+        orig_mod = state["_orig_mod"]
+        self.__class__ = orig_mod.__class__
```

Comment:
This will make this object not be an OptimizedModule anymore. That sounds bad?

Reply:
So I realize this looks dumb but here's my reasoning. Ultimately a user is expecting the same nn module they tried to compile because they're going to eventually load a state dict where they expect the keys to match exactly. Changing the name of the nn module to _orig_mod seems wrong as well, I could instead rename things better but figured why not just return something that's not an optimized module. So far this seems safe with the exception of copy logic.

Reply:
An example of a user getting confused by the existing state dict https://discuss.pytorch.org/t/state-dict-after-compile-and-onnx-export/180137